src/main/kotlin/no/nav/familie/ef/iverksett/oppgave/OpprettFremleggsoppgaverTask.kt

```diff
@@ -0,0 +1,101 @@
+package no.nav.familie.ef.iverksett.oppgave
+
+import no.nav.familie.ef.iverksett.infrastruktur.task.opprettNestePubliseringTask
+import no.nav.familie.ef.iverksett.iverksetting.IverksettingRepository
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettBarnetilsyn
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettData
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettOvergangsstønad
+import no.nav.familie.ef.iverksett.iverksetting.domene.IverksettSkolepenger
+import no.nav.familie.ef.iverksett.iverksetting.domene.OppgaverForOpprettelse
+import no.nav.familie.ef.iverksett.repository.findByIdOrThrow
+import no.nav.familie.kontrakter.ef.iverksett.OppgaveForOpprettelseType
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.prosessering.internal.TaskService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+@TaskStepBeskrivelse(
+    taskStepType = OpprettFremleggsoppgaverTask.TYPE,
+    beskrivelse = "Oppretter fremleggsoppgave for overgangsstønad og barnetilsyn",
+)
+class OpprettFremleggsoppgaverTask(
+    private val oppgaveService: OppgaveService,
+    private val iverksettingRepository: IverksettingRepository,
+    private val taskService: TaskService,
+) : AsyncTaskStep {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    override fun onCompletion(task: Task) {
+        taskService.save(task.opprettNestePubliseringTask())
+    }
+
+    companion object {
+        const val TYPE = "opprettFremleggsoppgaver"
+    }
+
+    override fun doTask(task: Task) {
+        val behandlingId = UUID.fromString(task.payload)
+        val iverksett = iverksettingRepository.findByIdOrThrow(behandlingId)
+        if (iverksett.data is IverksettSkolepenger) {
+            logger.info(
+                "Oppretter ikke fremleggsoppgave for behandling=$behandlingId" +
+                    " da det er en skolepenger (${iverksett::class.java.simpleName})",
+            )
+            return
+        }
+
+        opprettFremleggsoppgaveHvisOppgavetypeFinnes(OppgaveForOpprettelseType.INNTEKTSKONTROLL_1_ÅR_FREM_I_TID, iverksett.data, behandlingId)
+        opprettFremleggsoppgaveHvisOppgavetypeFinnes(OppgaveForOpprettelseType.INNTEKTSKONTROLL_SELVSTENDIG_NÆRINGSDRIVENDE, iverksett.data, behandlingId)
```

Comment:
Man kan kanskje få det litt mer lesbart ved å unngå to kall her, når enums uansett er hardkodet.